sources/core/Stride.Core.Mathematics/Half2.cs

```diff
@@ -229,5 +229,16 @@ public override bool Equals(object obj)
         {
             return new Vector2(value.X, value.Y);
         }
+                                
+        /// <summary>
+        /// Deconstructs the vector's components into named variables.
+        /// </summary>
+        /// <param name="x">The X component</param>
+        /// <param name="y">The Y component</param>
+        public void Deconstruct(out float x, out float y)
```

Comment:
I'm not sure about deconstructing it onto `float` instead of `Half`. We should preserve the data type precisely if possible.